File: backend/AssignmentSubmissionSystem.API/Models/Subject.cs
using System.ComponentModel.DataAnnotations;
namespace AssignmentSubmissionSystem.API.Models;

public class Subject{
    public int Id {get;set;}
    [Required]
    [MaxLength(50)]
    public string Name {get;set;} = string.Empty;
    public int AcademicClassId {get;set;}
    public AcademicClass AcademicClass{get;set;} = null!;
    public ICollection<Assignment> Assignments {get;set;} = new List<Assignment>();
    public ICollection<TeacherAssignment> TeacherAssignments { get; set; }
    = new List<TeacherAssignment>();
}

File: frontend/assignmentsubmissionsystem.web/types/subject.ts
export interface Subject {
  id: number;
  name: string;

  academicClassId: number;
  academicClassName: string;
}

export interface CreateSubjectRequest {
  name: string;
  academicClassId: number;
}

export interface UpdateSubjectRequest {
  name: string;
  academicClassId: number;
}
